app/models/TraderProfile.scala

```diff
@@ -104,4 +104,26 @@ object TraderProfile {
       case Left(errors) => Left(errors)
     }
 
+  def validateHasNiphl(
```

Comment:
what is the difference between buildNiphl and validate Niphl - i feel we should be using buildNiphl and not separating it out into has and number

Reply:
We use validateHasNiphl because we are just validating the current data and we don't need to build a new TraderProfile which is what buildNiphl does, if that makes sense?

Reply:
it does - it's just build Niphl already does that - it only returns a built trader profile if it passes the tests - also we can use just one validation for both niphl number and has niphl

Reply:
what is validate niphl returning?

Reply:
you're right I've updated this now to use buildNiphl instead 😄